libs/platform/ebpf_epoch.c

```diff
@@ -372,3 +397,40 @@ _ebpf_flush_worker(void* context)
     }
     ebpf_epoch_flush();
 }
+
+ebpf_epoch_work_item_t*
+ebpf_epoch_allocate_work_item(void* callback_context, void (*callback)(void* context))
+{
+    ebpf_epoch_work_item_t* work_item = ebpf_allocate(sizeof(ebpf_epoch_work_item_t), EBPF_MEMORY_NO_EXECUTE);
+    if (!work_item) {
+        return NULL;
+    }
+
+    work_item->callback = callback;
+    work_item->callback_context = callback_context;
+    work_item->header.entry_type = EBPF_EPOCH_ALLOCATION_WORK_ITEM;
+
+    return work_item;
+}
+
+void
+ebpf_epoch_schedule_work_item(ebpf_epoch_work_item_t* work_item)
```

Comment:
When and why will a WI be scheduled on epoch end?

Reply:
Code needs to delete the pages allocated to hold code. We can't use the normal memory allocator as it isn't page aligned.

Reply:
Please add a // comment somewhere relevant explaining that.